### PR TITLE
completed task 8

### PR DESCRIPTION
**Pull request by jamesraymetcalfe**

### CODE REVIEW

File: models/articles-model.js [hunk 1]
```diff
@@ -26,3 +26,23 @@ exports.selectArticlesByID = (articles_id) => {
       return data.rows[0];
     });
 };
+
+exports.updateArticleByID = (inc_votes, articles_id) => {
+  if (inc_votes === 0 || inc_votes < -1 || inc_votes > 1 ) {
```
Comment (nc-hannah-burks) [inline models/articles-model.js:31]: We should actually be able to increment our votes by more than 1, and also use this to decrement the votes. So ideally we want to remove this check.

File: models/articles-model.js [hunk 2]
```diff
@@ -26,3 +26,23 @@ exports.selectArticlesByID = (articles_id) => {
...
+    .query(
+      `UPDATE articles
+  SET votes = votes + $1
+  WHERE article_id = $2
+  RETURNING *;`,
+      [inc_votes, articles_id]
+    )
+    .then((data) => {
+      if (data.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "article does not exist" });
```
Comment (nc-hannah-burks) [inline models/articles-model.js:44]: Nice error handling!

File: __tests__/app.tests.js [hunk 3]
```diff
@@ -74,37 +107,75 @@ describe("/api/articles/:articles_id", () => {
...
+    const vote = { inc_votes: 1 };
+    return request(app)
+      .patch("/api/articles/forklift")
+      .send(vote)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("bad request");
+      });
+  });
+  test("PATCH:400 sends an appropriate status and error when the vote value is 0", () => {
```
Comment (nc-hannah-burks) [inline __tests__/app.tests.js:151]: This should actually be acceptable, it just shouldn't amend the votes.

File: __tests__/app.tests.js [hunk 4]
```diff
@@ -74,37 +107,75 @@ describe("/api/articles/:articles_id", () => {
...
+    const vote = { inc_votes: "forklift" };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(vote)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("bad request");
+      });
+  });
+  test("PATCH:400 sends an appropriate status and error message when the request is more than a single vote", () => {
```
Comment (nc-hannah-burks) [inline __tests__/app.tests.js:171]: This should also be acceptable, I would actually recommend adding another couple of Status 200 tests for if inc_votes is 0, inc_votes is a negative number and inc_votes is a larger number than 1

File: __tests__/app.tests.js [hunk 5]
```diff
@@ -74,37 +107,75 @@ describe("/api/articles/:articles_id", () => {
         expect(response.body.msg).toBe("bad request");
       });
   });
-});
-
-describe("/api/articles", () => {
```
Comment (nc-hannah-burks) [inline __tests__/app.tests.js:79]: You might also look to add a test for 
Status 200, if inc_votes is missing, should not update article and return original single article object (this may be included as a 400 but that could make for a brittle test suite)
